bitcoin/src/blockdata/script.rs

```diff
@@ -170,14 +170,54 @@ pub enum Error {
     Serialization
 }
 
+// If bitcoinonsensus-std is off but bitcoinconsensus is present we patch the error type to
+// implement `std::error::Error`.
+#[cfg(all(feature = "std", feature = "bitcoinconsensus", not(feature = "bitcoinconsensus-std")))]
+mod bitcoinconsensus_hack {
+    use core::fmt;
+
+    #[repr(transparent)]
+    pub(crate) struct Error(bitcoinconsensus::Error);
+
+    impl fmt::Debug for Error {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            fmt::Debug::fmt(&self.0, f)
+        }
+    }
+
+    impl fmt::Display for Error {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            fmt::Display::fmt(&self.0, f)
+        }
+    }
+
+    // bitcoinconsensus::Error has no sources at this time
+    impl std::error::Error for Error {}
+
+    pub(crate) fn wrap_error(error: &bitcoinconsensus::Error) -> &Error {
+        // Unfortunately, we cannot have the reference inside `Error` struct because of the 'static
+        // bound on `source` return type, so we have to use unsafe to overcome the limitation.
+        // SAFETY: the type is repr(transparent) and the lifetimes match
+        unsafe {
+            &*(error as *const _ as *const Error)
```

Comment:
Could you do a `transmute<&'a bitcoinconsensus::Error, &'a Error>` so that it's clear that the lifetimes match and that you're just turning one kind of pointer into another?

Reply:
AFAIK pointer casts are preferred to transmutes by Rust community but I will investigate.

Reply:
A transmute would be safer because you can force transmutes not to change lifetimes.

Reply:
The signature is already doing that, though.

Reply:
Sure, but the signature is at the function level. The transmute controls the lifetime at the point of the transmute call. So if someone were to change the function signature for some reason to something that'd be unsafe, or if thte result of the transmute were used as an intermediate value in an illegal way, then the transmute would stop compiling.